### ISSUE TITLE
consistent-function-scoping false positive calling same-scoped function

### ISSUE THREAD

**Issue by kevinoid** [sindresorhus/eslint-plugin-unicorn#375]
Running ESLint 6.4.0 with eslint-plugin-unicorn 11.0.0 and "extends": "plugin:unicorn/recommended" on a file with the following reduced example code:

```js
'use strict';

module.exports = function recordErrors(eventEmitter, stateArgument) {
  function onError(error) {
    stateArgument.inputError = error;
  }
  function onError2(error) {
    onError(error);
  }

  eventEmitter.once('error', onError2);
};
```
gives the following output:

    /tmp/consistent-function-scoping-issue/issue2.js
      7:3  error  Move function to the outer scope  unicorn/consistent-function-scoping
    
    ✖ 1 problem (1 error, 0 warnings)

I believe this is incorrect since `onError2` can not be moved to the global scope, since it calls `onError`, which is only defined in the function scope.

Thanks,
Kevin

Note: Split from #374 per https://github.com/sindresorhus/eslint-plugin-unicorn/issues/374#issuecomment-532348080

**Comment by kevinoid** [comment]
Thanks for the very quick fix @MrHen!

**Comment by kevinoid** [comment]
My apologies for the late follow-up, but I don't believe this issue is completely fixed.

With ESLint 6.8.0, eslint-plugin-unicorn 15.0.1, and "extends": "plugin:unicorn/recommended" I still observe the same error with the original example code.  #378 appears to have fixed the issue if `module.exports = ` is not present, but when it is present (as in the original example) the issue still occurs.

Thanks,
Kevin
